crates/forge_services/src/truncator.rs

```diff
@@ -0,0 +1,231 @@
+#![allow(dead_code)]
+/// Maximum character limit for truncation
+const MAX_LIMIT: usize = 40_000;
+
+/// Result of a truncation operation
+#[derive(Debug, Clone, PartialEq)]
+pub struct TruncationResult {
+    /// The actual content passed for truncation.
+    pub actual: String,
```

Comment:
```suggestion
    pub actual: &'a str,
```
Can improve performance here.